cloudnet/src/main/java/de/dytanic/cloudnet/console/JLine2Console.java

```diff
@@ -15,16 +17,53 @@
 
     private String screenName = version;
 
+    private AbstractConsoleAnimation runningAnimation;
+    private Thread currentAnimationThread;
+
     public JLine2Console() throws Exception {
         AnsiConsole.systemInstall();
 
         this.consoleReader = new ConsoleReader();
         this.consoleReader.setExpandEvents(false);
     }
 
+    @Override
+    public AbstractConsoleAnimation getRunningAnimation() {
+        return this.runningAnimation;
+    }
+
+    @Override
+    public void startAnimation(AbstractConsoleAnimation animation) {
+        Validate.checkNotNull(animation);
+        if (this.runningAnimation != null) {
+            throw new IllegalStateException("Cannot run multiple animations at once");
+        }
+
+        this.runningAnimation = animation;
+
+        if (this.currentAnimationThread != null) {
+            this.currentAnimationThread.interrupt();
+        }
+
+        animation.setConsole(this);
+
+        this.currentAnimationThread = new Thread(() -> {
```

Comment:
Don't create a thread for this. A ThreadPool with pool size one would be good for this